Unit Tests For Event Service and DB Mocking

Jamie-Unosquare opened the pull request:
Added Functionality to allow EntityFrameworkContext and Repository Mocking
String interpolation and removal of unused method from Event Service
Added missing info from MockDatabase

Jamie-Unosquare committed:
Unit Tests For Event Service and DB Mocking

Added Functionality to allow EntityFrameworkContext and Repository Mocking
String interpolation and removal of unused method from Event Service
Added missing info from MockDatabase

Jamie-Unosquare commented:
Currently a lot of the expected results are just written into the asserts as they're set up in the MockDatabase class. Would like opinions on if this is fine or if I should add a variable that queries the mockdatabase to get the actual number for example querying all events for annualLeave to compare for the necessary test.

BrownMark commented:
To be honest Jamie, what you're mentioning is my reservation with the mock database approach. I personally prefer the arrangements for a unit test to happen in the unit test. Makes maintenance much more straightforward and less likely to break other unit tests.

Jamie-Unosquare commented:
Do you have any suggestions to what would be the best way to do this?

Jamie-Unosquare committed:
Removed Singular Set up MockDB Method
DB repositories are spun up for indivual tests when necessary
TestClassBuilder for easy class non-customised building for quick unit testing
Extended the mapper to include a missing mapping

Jamie-Unosquare commented on AdminCore.Services.Tests/MockDatabase.cs:
```diff
@@ -17,6 +17,7 @@ public class MockDatabase
     public List<EventStatus> EventStatusRepository;
     public List<EventType> EventTypeRepository;
     public List<Team> TeamRepository;
+    public List<MandatoryEvent> MandatoryEventRepository;
```
These changes can be ignored for now as this class will be deleted when DashboardService tests are updated

Jamie-Unosquare committed:
Test to cover remaining lines missing from coverage
